File: Belajar_Day_3/Day3_OOP4.java

public class Day3_OOP4{
    public static void main(String[] args) {
        Customer customer1 = new Customer("Rina",27, 10000000, 3000000);
        customer1.showSummary();
        double dbr1 = customer1.hitungBDR();
        System.out.println("DBR: " + dbr1 + "%");
        
        //karena ga pake static, harus new
        //biar bisa buat object, jangan pakai static
        RiskAssessor risk1 = new RiskAssessor();
        String decision = risk1.assess(customer1);
        System.out.println(decision);
    }
}

class Customer {
        String name;
        int age;
        double salary;
        float existingLoan;

        //constructors
        public Customer (String name, int age, double salary, float existingLoan){
            this.name = name;
            this.age = age;
            this.salary = salary;
            this.existingLoan = existingLoan;
        }

        //behaviour/actions/method
        double hitungBDR(){
            double dbr = (this.existingLoan/this.salary) * 100;
            return dbr;
        }

        public void showSummary(){
            System.out.println("NAMA: " + this.name);
            System.out.println("AGE: " + this.age);
            System.out.println("SALARY: " + this.salary);
            System.out.println("EXISTING LOAN: " + this.existingLoan);
            //System.out.println("DBR: " + hitungBDR() + "%");
            
    } 
}

class RiskAssessor{
    public String assess(Customer customer){
        String decision;
        if(customer.age < 21){
            decision = "Not eligible";
        } else if (customer.salary >= 8000000){
            decision = "Eligible";
        } else if (customer.salary >= 3000000){
            decision = "Manual review";
        } else {
            decision = "Not eligible";
        }
        return decision;
    }
}

File: Belajar_Day_3/Day3_OOP4.py
#buat attribute baru: existing loan (float)
#buat method baru: BDR (existing_loan/salary) * 100% -> return float
class Customer:
    def __init__(self, name , age, salary,existing_loan):
        self.name = name
        self.age = age
        self.salary = salary
        self.existing_loan = existing_loan

    def show_summary(self):
        print("Customer:", self.name)
        print("Age:", self.age)
        print("Salary:", self.salary)
        print("Existing loan:", self.existing_loan)

    def hitung_dbr(self):
        dbr = (self.existing_loan/self.salary) * 100
        return dbr
    
class RiskAssessor:
    def assess(self,customer):
        if customer.age < 21:
            return "Not Eligible"
        elif customer.salary >= 8000000:
            return "Eligible"
        elif customer.salary >= 3000000:
            return "Manual Review"
        else:
            return "Not eligible"
        
customer1 = Customer("Rina", 27, 7000000, 3000000)
customer1.show_summary()
print("dbr:", customer1.hitung_dbr(), "%")


assessor = RiskAssessor()
print(assessor.assess(customer1))